src/index.js

```diff
@@ -1,15 +1,60 @@
 import React from "react";
 import ReactDOM from "react-dom";
-import "./index.css";
-import App from "./App";
 import reportWebVitals from "./reportWebVitals";
+import axios from "axios";
 
+import "./index.css";
+import App from "./App";
+import UserUtils from "./components/shared/user";
 import * as themes from "./theme/schema";
 import { setToLS } from "./utils/storage";
 
+// Add a request interceptor
+axios.interceptors.request.use(
+  (config) => {
+    const token = UserUtils.getAccessToken();
+    if (token) {
+      config.headers["Authorization"] = "Bearer " + token;
+    }
+    // config.headers['Content-Type'] = 'application/json';
+    return config;
+  },
+  (error) => {
+    Promise.reject(error);
+  }
+);
+
+axios.interceptors.response.use(
+  (response) => {
+    return response;
+  },
+  function (error) {
+    const originalRequest = error.config;
+    if (error.response.status === 401 && !originalRequest._retry) {
+      originalRequest._retry = true;
+      return axios
+        .post("http://127.0.0.1:8000/auth/refresh", {
```

Comment:
the base URL should be placed somewhere in the app that's accessible everywhere instead of hardcoding it